runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -310,12 +343,69 @@ void chpl_mem_layerInit(void) {
   //   jemalloc 4.5.0 man: "Once, when the first call is made to one of the
   //   memory allocation routines, the allocator initializes its internals"
   if (heap_base != NULL) {
-    heap.base = heap_base;
-    heap.size = heap_size;
-    heap.cur_offset = 0;
-    if (pthread_mutex_init(&heap.alloc_lock, NULL) != 0) {
-      chpl_internal_error("cannot init chunk_alloc lock");
+    int hpi;
+
+    //
+    // For now, we only NUMA-localize the comm layer desired shared heap
+    // if we're using the ugni comm layer.  ugni is the simpler case
+    // because it's only used on Cray X* compute blades which are a
+    // real-memory architecture.  We'll tackle comm=gasnet later; that's
+    // more complicated because GASNet is used on all sorts of systems,
+    // some virtual-memory, and we can't just go touching all the pages.
+    //
+    if (strcmp(CHPL_COMM, "ugni") != 0
```

Comment:
I would prefer to explicit exclude flat here. I think we'll end up excluding it currently because chpl_topo_getNumNumaDomains() will be zero under flat currently, but that might not always be the case